### PR TITLE
fix(android): fix firing `PlaybackTrackChanged` event in with single track in `RepeatMode.Queue`

### LINKED ISSUE #1612: Issue with `PlaybackTrackChanged` event on Android with single track queue in `RepeatMode.Queue`
This issue represents the uncompleted work from https://github.com/doublesymmetry/react-native-track-player/pull/1501#issuecomment-1110096083. A decision was made to merge what we had as it solved a large number of issues and deal with the rest on a dedicated issue.

> Remaining Issue:
> 
> In a single-track queue with RepeatMode.Queue Android on "Skip" no track-changed is fired.
> 
> Should be fired with nextTrack, position, and track (according to iOS).
> Natural track end works correctly.

### PR DESCRIPTION
**Pull request by mpivchev**

### CODE REVIEW

**Comment by jspizziri** [comment]
@mpivchev the commit history on this branch seems a little weird. I'm guessing you branched off of a branch that has since been merged & squashed?

File: android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt [hunk 1]
```diff
@@ -340,14 +340,15 @@ class MusicService : HeadlessJsTaskService() {
...
 
-                if (it == AudioPlayerState.ENDED && player.nextItem == null) {
-                    Bundle().apply {
-                        putInt(TRACK_KEY, player.currentIndex)
-                        putDouble(POSITION_KEY, player.position.toSeconds())
+                Bundle().apply {
+                    putInt(TRACK_KEY, player.currentIndex)
+                    putDouble(POSITION_KEY, player.position.toSeconds())
 
+                    if (it == AudioPlayerState.ENDED && player.nextItem == null) {
```
[the remark this answers is not in the document]
Comment (jspizziri) [reply android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt:347]: @mpivchev this looks good. One small thing: can we move the emit of the queue-ended to be _after_ the track changed? I know they're async so there's no guarantee, but I'm guessing more times than not they'll arrive in order and this is the ordering of iOS (track changed first, then queue ended)

[the remark this answers is not in the document]
Comment (mpivchev) [reply android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt:349]: Done :)

**Review by jspizziri — CHANGES_REQUESTED** [review]
@mpivchev something doesn't seem quite right with this fix. 

My testing shows that the stock `example` app emits the following events when the app loads:

```
 LOG  Event.PlaybackTrackChanged {"position": 0, "track": 0}
 LOG  Event.PlaybackTrackChanged {"nextTrack": 0, "position": 0}
 LOG  Event.PlaybackTrackChanged {"position": 0, "track": 0}
 LOG  Event.PlaybackTrackChanged {"position": 0, "track": 0}
 ```
 
 This is prior to any interaction with the queue, only adding tracks. It seems that conditional was preventing duplicate events from being fired.

**Comment by mpivchev** [comment]
That's unfortunate. Will look into another solution

**Comment by mpivchev** [comment]
@jspizziri Do these duplicate events produce any unwanted results? I think it's ok to leave them firing if that's the case. But welcome to discuss this.

**Comment by jspizziri** [comment]
@mpivchev users are keying off of these events to do various things in their apps. If there are duplicates I'd imagine it will certainly produce unwanted results for them, yes.

**Comment by jspizziri** [comment]
I think this can be closed with the new events structure introduced by https://github.com/doublesymmetry/react-native-track-player/pull/1713

**Pull request closed without merging**